lab_1/main.py

```diff
@@ -2,40 +2,62 @@
 Lab 1
 A concordance extraction
 """
+
 text = open('data.txt', encoding = 'utf-8').read()
 
-# ФУНКЦИЯ РАБОТАЕТ, НО ПОСМОТРЕТЬ, ПОЧЕМУ  ТЕСТЫ НЕ ВИДЯТ ФАЙЛ "TOKENS.TXT" + '`the'
-def tokenize(text):
+
+def tokenize(text: str) -> list:
+    """
+    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
+    :param text: the initial text
+    :return: a list of lowercased tokens without punctuation
+    e.g. text = 'The weather is sunny, the man is happy.'
+    --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
+    """
     import re
     if type(text) == str:
         new_text = ''
         for elem in text:
             if elem.isalpha() or re.fullmatch('[\s]', elem):
                 new_text += elem
-        new_text = new_text.lower()
-        new_text = new_text.split()
+        new_text = new_text.lower().split()
         return new_text
     else:
         return []
 
 list_of_tokens = tokenize(text)
 
-
 stop_words = open('stop_words.txt').read()
 stop_words = stop_words.split()
 
 
-# ФУНКЦИЯ РАБОТАЕТ ИДЕАЛЬНО!!!
-def remove_stop_words(token_list, stop_words):
+def remove_stop_words(tokens: list, stop_words: list) -> list:
+    """
+    Removes stop words
+    :param tokens: a list of tokens
+    :param stop_words: a list of stop words
+    :return: a list of tokens without stop words
+    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
+    stop_words = ['the', 'is']
+    --> ['weather', 'sunny', 'man', 'happy']
+    """
     new_list = [token for token in token_list if token not in stop_words]
     return new_list
 
 token_list = remove_stop_words(list_of_tokens, stop_words)
 
 
-def calculate_frequencies(token_list):
+def calculate_frequencies(tokens: list) -> dict:
+    """
+    Calculates frequencies of given tokens
+    :param tokens: a list of tokens without stop words
+    :return: a dictionary with frequencies
+    e.g. tokens = ['weather', 'sunny', 'man', 'happy']
+    --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
+    """
     if type(token_list) == list:
         dic = {token: token_list.count(token) for token in token_list}
+        print(dic)
         return dic
     else:
```

Comment:
you don't need else here
if "return dic" works, the function will stop its work